src/codewhisperer/util/editorContext.ts

```diff
@@ -109,12 +109,21 @@ export async function buildListRecommendationRequest(
 
     logSupplementalContext(supplementalContexts)
 
+    const supplementalContext: codewhispererClient.SupplementalContext[] = supplementalContexts
+        ? supplementalContexts.contents.map(v => {
+              return {
+                  filePath: v.filePath,
+                  content: v.content,
+              }
+          })
+        : []
+
```

Comment:
oh because here I need to return a different pojo which is `codewhispererClient.SupplementalContext`
```
{
  content: string,
  filePath: string
} 

```
where as my plugin Pojo has 3 fields (1 extra)
```
{
   content: string,
   filePath: string,
  score: number
}
```

Reply:
is there an easier cleaner syntax to achieve this purpose?

Reply:
They're just called "objects" :)

You can use `selectFrom` in `src/shared/utilities/tsUtils.ts`:
```ts
const supplementalContext = supplementalContexts?.map(x => selectFrom(x, 'content', 'filePath') ?? []
```

Reply:
gotchaa